libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -675,102 +469,36 @@ void AC_AutoTune::control_attitude()
-                break;
-            case YAW:
-                updating_rate_d_down(tune_yaw_rLPF, AUTOTUNE_RLPF_MIN, AUTOTUNE_RD_STEP, tune_yaw_rp, AUTOTUNE_RP_MIN, AUTOTUNE_RP_MAX, AUTOTUNE_RP_STEP, target_rate, test_rate_min, test_rate_max);
-                break;
-            }
+            updating_rate_d_down_all(axis);
             break;
         // Check results after mini-step to increase rate P gain
         case RP_UP:
-            switch (axis) {
-            case ROLL:
-                updating_rate_p_up_d_down(tune_roll_rd, min_d, AUTOTUNE_RD_STEP, tune_roll_rp, AUTOTUNE_RP_MIN, AUTOTUNE_RP_MAX, AUTOTUNE_RP_STEP, target_rate, test_rate_min, test_rate_max);
-                break;
-            case PITCH:
-                updating_rate_p_up_d_down(tune_pitch_rd, min_d, AUTOTUNE_RD_STEP, tune_pitch_rp, AUTOTUNE_RP_MIN, AUTOTUNE_RP_MAX, AUTOTUNE_RP_STEP, target_rate, test_rate_min, test_rate_max);
-                break;
-            case YAW:
-                updating_rate_p_up_d_down(tune_yaw_rLPF, AUTOTUNE_RLPF_MIN, AUTOTUNE_RD_STEP, tune_yaw_rp, AUTOTUNE_RP_MIN, AUTOTUNE_RP_MAX, AUTOTUNE_RP_STEP, target_rate, test_rate_min, test_rate_max);
-                break;
-            }
+            updating_rate_p_up_all(axis);
             break;
         // Check results after mini-step to increase stabilize P gain
         case SP_DOWN:
-            switch (axis) {
-            case ROLL:
-                updating_angle_p_down(tune_roll_sp, AUTOTUNE_SP_MIN, AUTOTUNE_SP_STEP, target_angle, test_angle_max, test_rate_min, test_rate_max);
-                break;
-            case PITCH:
-                updating_angle_p_down(tune_pitch_sp, AUTOTUNE_SP_MIN, AUTOTUNE_SP_STEP, target_angle, test_angle_max, test_rate_min, test_rate_max);
-                break;
-            case YAW:
-                updating_angle_p_down(tune_yaw_sp, AUTOTUNE_SP_MIN, AUTOTUNE_SP_STEP, target_angle, test_angle_max, test_rate_min, test_rate_max);
-                break;
-            }
+            updating_angle_p_down_all(axis);
             break;
         // Check results after mini-step to increase stabilize P gain
         case SP_UP:
-            switch (axis) {
-            case ROLL:
-                updating_angle_p_up(tune_roll_sp, AUTOTUNE_SP_MAX, AUTOTUNE_SP_STEP, target_angle, test_angle_max, test_rate_min, test_rate_max);
-                break;
-            case PITCH:
-                updating_angle_p_up(tune_pitch_sp, AUTOTUNE_SP_MAX, AUTOTUNE_SP_STEP, target_angle, test_angle_max, test_rate_min, test_rate_max);
-                break;
-            case YAW:
-                updating_angle_p_up(tune_yaw_sp, AUTOTUNE_SP_MAX, AUTOTUNE_SP_STEP, target_angle, test_angle_max, test_rate_min, test_rate_max);
-                break;
-            }
+            updating_angle_p_down_all(axis);
```

Comment:
updating_angle_p_up_all()?

Reply:
Wasn't sure how to do this.  the tradheli version of updating_ methods uses different arguments.  And they are called from a common method in the autotune library.  I know that I can have separate versions of the method containing separate arguments but then I would have to know which frame type it is to know which method to use.